### ISSUE TITLE
Feature Request: The ability to extract the No-Intro sets.

### ISSUE THREAD

**Issue by TomTurbine** [Xpl0itU/WiiUDownloader#78]
With the official CDNs set to shut down next year, the No-Intro group has largely datted the entire CDN at this point and each of them includes the tmds even versioned as well.

Basically having an application that would be able to extract them as needed from the set even being able to extract the different version of the updates and all would be a welcome thing.

**Comment by coreyemtp** [comment]
I disagree, I think it's No-Intro that needs to fix their DAT. It should be 1:1 with what is downloaded from the CDN, but it's a complete mess right now.

**Comment by TomTurbine** [comment]
> I disagree, I think it's No-Intro that needs to fix their DAT. It should be 1:1 with what is downloaded from the CDN, but it's a complete mess right now.

If they just did a file name match, it would either be missing a ton of files as they have multiple versions of updates that just build off the previous updates or you would have massive amounts of repeating files more than doubling the size of the set because of it as they would have multiple full updates of the exact same game sometimes.

Remember how many Smash Brothers updates they have, no imagine having to store the full version of each one of them even though you know that every file of the older updates are also in the newer ones.

The way they have it setup allows the set to have each version of each update in it. It is the equivalent of having a merged Mame set versus of non-merged set.

And it isn't a mess, they have an actual scheme to how the files are named.

**Comment by coreyemtp** [comment]
Agreed, of course a simple filename match would be useless, but I don't really see the relevance since CRC32/SHA1 is used.

Ultimately there is a final version of each game, and they will never see an (official) update. Therefore, the definitive set would be the main game and the final update. That is what is downloaded when you request the game. I understand you can download a prior version, but that is a niche use for the "maybe I'll need it for a hack some day" crowd or those who inexplicably believe that every byte of data generated deserves preservation. That falls more in line with the TOSEC/GoodSets way of thinking.  Don't muck up the main DAT files with a bunch of garbage that is useless to 99.9% of users.

**Comment by TomTurbine** [comment]
> Agreed, of course a simple filename match would be useless, but I don't really see the relevance since CRC32/SHA1 is used.
> 
> Ultimately there is a final version of each game, and they will never see an (official) update. Therefore, the definitive set would be the main game and the final update. That is what is downloaded when you request the game. I understand you can download a prior version, but that is a niche use for the "maybe I'll need it for a hack some day" crowd or those who inexplicably believe that every byte of data generated deserves preservation. That falls more in line with the TOSEC/GoodSets way of thinking. Don't muck up the main DAT files with a bunch of garbage that is useless to 99.9% of users.

Not entirely true. With the current setup, it allows you to keep virtually every update with minimal space as the last update is built off the older updates anyways. And if they exclude the older stuff, then it isn't a proper CDN set. That's like asking them to start purging all the older revisions of SNES games as well as their betas and prototypes.

Sure, most people won't use them but that still means its an incomplete set in the DAT file. 

Then it comes to games like Smash Brothers where they change balance from patch to patch where some players liked it before.

Now, don't get me wrong, I also believe that 99.9% of all users will only care about the last update, but that isn't what the No-Intro group is about doing. They are trying to preserve it all in their sets and were actually debating on whether they should include every single aftermarket and homebrew app in their sets.

Just glad they finally came to the decision to split those dats in the near future where all the official, unlicensed games, and virtual console/mini-console games get one dat for that system and then all the aftermarket and homebrew will go into another. Will leave one relatively stable platform of the actually released stuff and one constantly updated with every homebrew and aftermarket release in their collection.

**Comment by coreyemtp** [comment]
We definitely agree on the latter part, splitting the DATs is much needed!
________________________________

**Comment by TomTurbine** [comment]
Any words on this?

The title.tmd is literally just the tmd files in the dat with the number being the version of the tmd it belongs to and the files without extension is just needs the app extension and then the tickets generated and could just use the games as they are.

I literally just downloaded stuff using this app and ran them through the dats and they all matched up except the files that the app can already generate on its own.

Adding No-Intro support would be great and would give this app usability long beyond when the servers don't even allow you access to them anymore and would solve the occasional issues where the files can't be extracted because something is going wrong when it is being downloaded from the servers which I have had happen a few times. 

With this, if it matches the No-Intro, you know its  correct so just extract away. Who knows, maybe No-Intro could use this as the basis for an unofficial decrypted set later on.
